test/bdd/agent_controller_steps.go

```diff
@@ -0,0 +1,211 @@
+/*
```

Comment:
I'm guessing we will need a TODO to create multiple BDD tests rather than having everything at the top level.

Reply:
Created #584 and added TODO